helpers.c

```diff
@@ -26,9 +26,19 @@ int sync_file_range(int fd, uint64_t offset, uint64_t nbytes,
 }
 #endif
 
+#if defined (__APPLE__)
+int posix_fadvise(int fd, off_t offset, off_t len, int advice)
+{
+	if (advice == POSIX_FADV_DONTNEED)
+		return discard_pages(fd, offset, len);
+        errno = EINVAL; 
```

Comment:
You can't see it in GitHub but your file has trailing whitespace in it (I can see it when I look at `git log -p` or `git diff HEAD^` - can you remove all instances of this?